src/streaming-main.cpp

```diff
@@ -151,10 +151,11 @@ static Result<NonnullRefPtr<JS::Program>, TestError> parse_harness_files(StringV
 
 static Result<void, TestError> run_test(StringView source, JS::Program::Type program_type, Vector<StringView> const& harness_files)
 {
+    auto program_or_error = parse_program(source, program_type);
```

Comment:
This change means we let a parse error take priority over a harness error.
I made this a separate commit in case it is unwanted since it is different behavior from the previous runner.
My opinion:
Since when we fail to parse we cannot run the test anyway it makes no sense to run the harness files. And thus if those fail as well it's not really going wrong in the harness files.